UnityProject/Assets/IngameDebugConsole/Scripts/DebugLogUnitystationCommands.cs

```diff
@@ -709,5 +699,60 @@ public static void AddAdmin(string userIDToPromote)
 
 			PlayerList.Instance.ProcessAdminEnableRequest(ServerData.UserID, userIDToPromote);
 		}
+
+		private static IEnumerator KillLights()
+		{
+			if (MatrixManager.MainStationMatrix?.Objects == null) yield break;
+
+			var currentIndex = 0;
+			var maximumIndexes = 20;
+			foreach (var stationObject in MatrixManager.MainStationMatrix.Objects.GetComponentsInChildren<LightSource>())
+			{
+				if (currentIndex >= maximumIndexes) yield return WaitFor.EndOfFrame;
+				stationObject.Integrity.ForceDestroy();
+				currentIndex++;
+			}
+		}
+
+		private static IEnumerator SelfPowerEverything()
+		{
+			if (MatrixManager.MainStationMatrix?.Objects == null) yield break;
+
+			var currentIndex = 0;
+			var maximumIndexes = 20;
+			foreach (var stationObject in MatrixManager.MainStationMatrix.Objects.GetComponentsInChildren<APCPoweredDevice>())
+			{
+				if (currentIndex >= maximumIndexes) yield return WaitFor.EndOfFrame;
+				stationObject.ChangeToSelfPowered();
+				currentIndex++;
+			}
+		}
+
+		public static bool IsAdmin()
+		{
+			if (AdminCommands.AdminCommandsManager.IsAdmin(PlayerManager.LocalPlayerScript.PlayerInfo.Connection, out _) != false) return true;
+			Logger.Log("This function can only be executed by admins.", Category.DebugConsole);
+			return false;
+		}
+
+		[ConsoleMethod("destroy-all-lights", "destroys all lights on the main station.")]
+		public static void DestroyAllLights()
+		{
+			if(IsAdmin() == false) return;
+			PlayerManager.LocalPlayerScript.StartCoroutine(KillLights());
```

Comment:
does this work just fine in a server - client configuration?

Reply:
good catch, I'll update it to run on the server.

Reply:
Done.